pkg/operation/botanist/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -171,32 +171,41 @@ else
   rm -f "{{ .pathBootstrapToken }}"
 fi
 
+# Try to find Node object for this machine if already registered to the cluster.
 NODENAME=
-ANNOTATION_RESTART_SYSTEMD_SERVICES="worker.gardener.cloud/restart-systemd-services"
+if [[ -s "{{ .pathNodeName }}" ]]; then
+  NODENAME="$(cat "{{ .pathNodeName }}")"
+elif [[ -f "{{ .pathKubeletKubeconfigReal }}" ]]; then
+  {{`NODENAME="$(`}}{{ .pathBinaries }}{{`/kubectl --kubeconfig="`}}{{ .pathKubeletKubeconfigReal }}{{`" get node -l "kubernetes.io/hostname=$(hostname | tr '[:upper:]' '[:lower:]')" -o go-template="{{ if .items }}{{ (index .items 0).metadata.name }}{{ end }}")"`}}
```

Comment:
Nit: just to make it obvious that we have a request for all Nodes with a label selector.

```suggestion
  {{`NODENAME="$(`}}{{ .pathBinaries }}{{`/kubectl --kubeconfig="`}}{{ .pathKubeletKubeconfigReal }}{{`" get nodes -l "kubernetes.io/hostname=$(hostname | tr '[:upper:]' '[:lower:]')" -o go-template="{{ if .items }}{{ (index .items 0).metadata.name }}{{ end }}")"`}}
```